base/set.jl

```diff
@@ -751,42 +778,65 @@ replace(a::Callable, b::Pair; count::Integer=-1) = throw(MethodError(replace, (a
 replace(a::Callable, b::Pair, c::Pair; count::Integer=-1) = throw(MethodError(replace, (a, b, c)))
 replace(a::AbstractString, b::Pair, c::Pair) = throw(MethodError(replace, (a, b, c)))
 
-
 ### replace! for AbstractDict/AbstractSet
 
 askey(k, ::AbstractDict) = k.first
 askey(k, ::AbstractSet) = k
 
-function _replace!(new::Callable, A::Union{AbstractDict,AbstractSet}, count::Int)
-    repl = Pair{eltype(A),eltype(A)}[]
+function _replace!(new::Callable, res::T, A::T,
+                   count::Int) where T<:Union{AbstractDict,AbstractSet}
     c = 0
-    for x in A
-        y = new(x)
-        if x !== y
-            push!(repl, x => y)
-            c += 1
+    if res === A # cannot replace elements while iterating over A
+        repl = Pair{eltype(A),eltype(A)}[]
+        for x in A
+            y = new(x)
+            if x !== y
+                push!(repl, x => y)
+                c += 1
+            end
+            c == count && break
+        end
+        for oldnew in repl
+            pop!(res, askey(first(oldnew), res))
```

Comment:
Very minor comment: as `res === A`, working with only one of `A` or `res` may make it slightly easier to read.